src/main/scala/org/hammerlab/guacamole/readsets/rdd/PartitionedRegions.scala

```diff
@@ -1,98 +1,214 @@
-    // will guide each pair to the designated partition in the `repartitionAndSortWithinPartitions` step below.
-    val keyedRegionRDDs: PerSample[RDD[(TaskPosition, R)]] =
-      regionRDDs.map(
-        _.flatMap(region => {
-          val singleContig = lociPartitionsBoxed.value.onContig(region.contigName)
-          val partitionsForRegion = singleContig.getAll(region.start - halfWindowSize, region.end + halfWindowSize)
-
-          // Update counters
-          totalRegions += 1
-          if (partitionsForRegion.nonEmpty) relevantRegions += 1
-          expandedRegions += partitionsForRegion.size
-
-          // Return this region, duplicated for each partition it is assigned to.
-          partitionsForRegion.map(partition => TaskPosition(partition, region.contigName, region.start) -> region)
-        })
-      )
+    val numPartitions = lociPartitioning.numPartitions
 
-    val numPartitions = lociPartitions.inverse.map(_._1).max + 1
+    progress(s"Partitioning reads according to loci partitioning:\n$lociPartitioning")
 
-    // Build RDDs of (read-set index, read) and merge these into one RDD,
-    sc
+    implicit val accumulableParam = new HistogramParam[Int, Long]
+
+    // Combine the RDDs, with each region keyed by its sample ID.
+    val keyedRegions: RDD[(SampleId, R)] =
+      sc
       .union(
         for {
-          (keyedRegionsRDD, sampleId) <- keyedRegionRDDs.zipWithIndex
+          (regionsRDD, sampleId) <- regionRDDs.zipWithIndex
         } yield {
           for {
-            (taskPosition, read) <- keyedRegionsRDD
+            region <- regionsRDD
           } yield
-            taskPosition -> (sampleId, read)
+            sampleId -> region
         }
       )
-      .repartitionAndSortWithinPartitions(KeyPartitioner(numPartitions))  // partition by TaskPosition.partition
-      .values  // Drop TaskPosition keys, leave behind just (read-set index, region) pairs.
+
+    // Histogram of the number of copies made of each region (i.e. when a region straddles loci-partition
+    // boundaries.
+    val regionCopiesHistogram: Accumulable[IntHist, Int] = sc.accumulable(IntHist(), "copies-per-region")
+
+    // Histogram of the number of regions assigned to each partition.
+    val partitionRegionsHistogram: Accumulable[IntHist, Int] = sc.accumulable(IntHist(), "regions-per-partition")
+
+    // Emit a copy of each region for each partition's alloted loci that it overlaps.
+    val partitionedRegions: RDD[(SampleId, R)] =
+      (for {
+        // For each region…
+        (sampleId, region) <- keyedRegions
+
+        // Partitions that should receive a copy of this region.
+        partitions = partitioningBroadcast.value.getAll(region, halfWindowSize)
+
+        // Update the copies-per-region histogram accumulator.
+        _ = (regionCopiesHistogram += partitions.size)
+
+        // For each destination partition…
```

Comment:
Weird, `git diff` to `less` works fine for me:

![](https://d3vv6lp55qjaqc.cloudfront.net/items/1U130i1O251d3r0e321P/Screen%20Shot%202016-08-10%20at%2012.59.07%20PM.png)

Happy to help you debug that if you want.

[This SO](http://serverfault.com/a/414763) suggests `export LESSCHARSET=utf-8`, but I don't have that set so it's definitely not a necessary condition.

[The next answer there](http://serverfault.com/a/414762) does list a few vars that I have set:

```bash
export LANG=en_US.UTF-8
export LANGUAGE=en_US.UTF-8
export LC_ALL="en_US.UTF-8"
```

So you could try setting those.